src/api/spec/controllers/webui/projects/public_key_controller_spec.rb

```diff
@@ -27,7 +27,8 @@
     context 'with a project that has no public key' do
       let(:keyinfo_response) { '<keyinfo />' }
 
-      it { expect(response.status).to eq(404) }
+      it { is_expected.to redirect_to(project_show_path(project)) }
+      it { expect(flash[:error]).not_to be_nil }
```

Comment:
You assume that we have to rewrite the such tests all the time. Which I doubt is true.

Besides, I am not arguing that we have to test everything in all detail. I am arguing that there is not much point in testing an error message without testing the content of the message.
So why not dropping that test?

> evanrolfe 10 minutes ago
> I think we should strive to write smart tests, not just lots of tests.
Exactly. And right now it seems we just add a test that allows us to say "We test the error message" which is vague enough that we never have to touch or bother with it again.

Sounds to me like a bad compromise.